Guide_1_Send_Payment.md

```diff
@@ -1,31 +1,112 @@
-- Send the payment
-- Track the payment progress manually
-- Track the payment progress automatically using subscriptions
+Each step is illustrated with Python code snippets, so you can execute each step as you read along. The snippets are ready-to-run programs and work in Python's interactive console.
 
 ## Prerequisites:
 Before you start, make sure you have the following things ready to go:
-- API credentials. Contact your Form3 contact to obtain your credentials. _(Note: Better name for this role? Who do they have to contact?)_
-- Register your bank ID and BIC with Form3
-- Sign up for Form3's scheme simulator. This is a sandbox environment that you can use to simulate transactions in order to test your application. _(Note: in the API this is called "payment simulator". Which term should we use?)_
+- API credentials. Contact Form3 to obtain your credentials.
+- Register your UK sortcodes and BICs with Form3.
+- Sign up for Form3's scheme simulator for FPS payments. This is a sandbox environment that you can use to simulate transactions in order to test your application.
+- If you want to run the Python code snippets, make sure you have [Python 2.7](https://www.python.org/downloads/) installed. You also need the requests package. The easiest way is to install it through [Pip](https://docs.python.org/2.7/installing/index.html): `pip install requests`
 
 
 ## Get a Bearer Token
 
-As a first step, you need to obtain a bearer token using your API credentials. This token will be used to authenticate yourself to the Form3 server.
+```python
+import requests
 
-_(Note: Should we elaborate on how to do that?)_
+### Replace these variables with your own data! ###
+client_id = 'YOUR CLIENT ID HERE'
+client_secret = 'YOUR CLIENT SECRET HERE'
+
+auth_payload = "grant_type=client_credentials"
+auth_url = 'https://api.tabla.env.form3.tech/v1/oauth2/token'
+auth_headers = {'Content-Type': 'application/x-www-form-urlencoded'}
+auth_request = requests.auth.HTTPBasicAuth(client_id, client_secret)
+
+auth = requests.request('post', auth_url, data=auth_payload, auth=auth_request, headers=auth_headers)
+print(auth.content)
+
+print("Bearer token: %s" % auth.json().get('access_token'))
+```
+
+As a first step, you need to obtain a bearer token using your client ID and your client secret. This token will be used in the other API calls to authenticate yourself to the Form3 server.
 
 ## Create the Payment
 
 To make a payment, you need to complete two steps. First create a payment resource to define the payment parameters, then send it by creating a payment submission resource. This payment submission is also used later to track the status of your payment.
 
 ### Create the Payment Resource
 
+```python
+import math, random, requests
+
+### Replace these variables with your own data! ###
+auth_token = 'A VALID BEARER TOKEN HERE'
+organisation_id = 'YOUR ORGANISATION ID HERE'
+bank_id = 'YOUR UK SORTCODE HERE'
+bank_id_code = 'YOUR BIC HERE'
```

Comment:
bank_id_code = GBDSC (thats the code that means the bank_id is a UK domestic sortcode). Might be worth some explanation of this during the introduction.
The bic isnt used during this tutorial (it is used for inbound payments though)

Reply:
I removed the variable and just use GBDSC in the payment attribute tree. Also added a sentence about it in the guide. What do you think?